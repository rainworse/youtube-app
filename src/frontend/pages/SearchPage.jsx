import VideoContainer from '../components/VideoContainer';
import APIFetcher from '../../backend/APIFetcher';
import ConfigHelper from '../../backend/ConfigHelper';
import { useParams } from 'react-router-dom';
import '../styles/SearchPage.css';

const SearchPage = () => {
  const params = useParams();
  const resultCount = ConfigHelper.getValue('search-page-video-count');

  return (
    <div className="search-results">
      <VideoContainer
        fetchMethod={APIFetcher.fetchSearchVideos}
        fetchParams={[params.q, resultCount]}
      />
    </div>
  );
};

export default SearchPage;
