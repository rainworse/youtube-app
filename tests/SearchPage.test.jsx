import { describe, expect, it } from 'vitest';
import UserContext from '../src/frontend/UserContext';
import { MemoryRouter } from 'react-router-dom';
import SearchPage from '../src/frontend/pages/SearchPage';
import { render, waitFor } from '@testing-library/react';

vi.mock('../src/backend/APIFetcher');

describe('SearchPage', () => {
  it('renders search page', async () => {
    const searchPage = render(
      <UserContext.Provider value={{ currentUser: 'user' }}>
        <MemoryRouter>
          <SearchPage />
        </MemoryRouter>
      </UserContext.Provider>
    );

    await waitFor(() => searchPage.getByText('test video4'));
    expect(searchPage.container).toMatchSnapshot();
  });
});
