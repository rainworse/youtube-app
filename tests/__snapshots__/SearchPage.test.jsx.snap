// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchPage > renders search page 1`] = `
<div>
  <div
    class="search-results"
  >
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/001"
      >
        <img
          alt="test video1"
          class="video-card-thumbnail"
          src="non/existent/location1"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video1
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel001"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel1
            </div>
          </a>
        </div>
      </a>
    </div>
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/002"
      >
        <img
          alt="test video2"
          class="video-card-thumbnail"
          src="non/existent/location2"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video2
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel002"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel2
            </div>
          </a>
        </div>
      </a>
    </div>
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/003"
      >
        <img
          alt="test video3"
          class="video-card-thumbnail"
          src="non/existent/location3"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video3
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel003"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel3
            </div>
          </a>
        </div>
      </a>
    </div>
    <div
      class="video-card"
    >
      <a
        class="video-card-thumbnail-anchor"
        href="/video/004"
      >
        <img
          alt="test video4"
          class="video-card-thumbnail"
          src="non/existent/location4"
        />
        <div
          class="video-card-text"
        >
          <div
            class="video-card-title"
          >
            test video4
          </div>
          <a
            class="video-card-channel-anchor"
            href="/channel/channel004"
          >
            <div
              class="video-card-channel"
            >
              by 
              test channel4
            </div>
          </a>
        </div>
      </a>
    </div>
  </div>
</div>
`;
